topydo/ui/columns/Main.py

```diff
@@ -269,36 +270,39 @@ def _execute_handler(self, p_command, p_todo_id=None, p_output=None):
 
         self._last_cmd = (p_command, p_output == self._output)
 
-        if '{}' in p_command:
-            if self._has_marked_todos():
-                p_todo_id = ' '.join(self.marked_todos)
-            p_command = p_command.format(p_todo_id)
-
         try:
             p_command = shlex.split(p_command)
         except ValueError as verr:
             self._print_to_console('Error: ' + str(verr))
             return
 
         try:
-            (subcommand, args) = get_subcommand(p_command)
+            subcommand, args = get_subcommand(p_command)
         except ConfigError as cerr:
             self._print_to_console(
                 'Error: {}. Check your aliases configuration.'.format(cerr))
             return
 
-        self._backup(subcommand, args)
+        cmd = lambda op: subcommand(op, self.todolist, p_output, self._output,
+                                    self._input)
+
+        if '{}' in args:
+            ids = self.marked_todos if self._has_marked_todos() else [p_todo_id]
+            transaction = Transaction(cmd, ids)
+        else:
+            transaction = Transaction(cmd)
+
+        transaction.prepare(args)
 
         try:
-            command = subcommand(
-                args,
-                self.todolist,
-                p_output,
-                self._output,
-                self._input,
-            )
+            self._backup(subcommand, args)
+        except TypeError:
+            self._print_to_console('Error: no todo item was marked for this'
```

Comment:
This error will be displayed if the user would enter in the commandline something like `do {}` without marking any items before (so the `p_todo_id` is `None` and `self._has_marked_todos()` returns `False`).